src/api/modelTypes.js

```diff
@@ -78,62 +78,25 @@ export type DevUser = $ReadOnly<{|
  * A Zulip user, which might be a human or a bot, as found in one realm.
  *
  * This is a user object as found in properties `realm_users` and
- * `realm_non_active_users` of a `/register` response.
- *
- * For details on the properties, see the Zulip API docs on `/users`:
- *   https://zulip.com/api/get-users#response
- * which returns almost the same set of properties.
- *
- * See also the comments on `UserProfile` in the server (lineno is approx.):
- *   https://github.com/zulip/zulip/blob/main/zerver/models.py#L734
- * Most properties correspond to fields on `UserProfile`, and many are
- * described most usefully there.
- *
- * For authoritative results, consult how `raw_users`, and then
- * `realm_users` and `realm_non_active_users`, are computed in
- * `zulip/zulip:zerver/lib/events.py` .
- *
- * Properties are listed below in the order they appear on `UserProfile`,
- * because that's the most logically-organized and also the most helpful
- * of the references above.
+ * `realm_non_active_users` of a `/register` response. See the API doc:
+ *   https://zulip.com/api/register-queue
  *
  * See also:
  *  * `CrossRealmBot` for another type of bot user, found in a
  *    different part of a `/register` response
  *  * `UserOrBot` for a convenience union of the two
  */
 export type User = $ReadOnly<{|
-  user_id: UserId,
-  email: string,
-
-  full_name: string,
-
-  date_joined: string,
-
-  // is_active doesn't appear in `/register` responses -- instead,
-  // users where is_active is true go in `realm_users`, and where false
-  // go in `realm_non_active_users`.  Shrug.
-
-  // is_admin corresponds to is_realm_admin in server code.
-  is_admin: boolean,
-
-  // is_guest included since commit d5df0377c (in 1.9.0); before that,
-  // there's no such concept, so effectively it's implicitly false.
-  is_guest?: boolean,
+  // Property ordering follows the doc.
+  // Current to feature level 121.
 
   // For background on the "*bot*" fields, see user docs on bots:
   //   https://zulip.com/help/add-a-bot-or-integration
   // Note that certain bots are represented by a different type entirely,
   // namely `CrossRealmBot`.
-  is_bot: boolean,
-  bot_type?: number,
-  bot_owner?: string,
```

Comment:
It sure is awkward that these three get scattered from each other. :-/ Also that `user_id` is in a random place in the middle instead of at the top, and that `is_guest` is nowhere near `is_admin` and `is_owner`.

I think the reordering is worth the trade-off in order to make it more tractable to bring the type up to date, but it's a shame that it actually gets harder to read.

Perhaps first reorder the API docs to match the more logical order that's in models.py and in this type? Then we don't have to regress on that.

Reply:
Replied [in chat](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/ordering.20of.20user.20properties/near/1371297).

Reply:
Sent #21883; I guess I'll await that and apply the result to my next revision.

Reply:
For my next revision, I'd appreciate the time saved by applying the new order on top of the existing commits in the branch, rather than having an early commit that takes the new ordering then resolve all the merge conflicts that'd create.

Reply:
(I've actually gone ahead and done in the proper way. 🙂)